docs/Plugin-BuildrootLock.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: default
+title: Plugin BuildrootLock
+---
+
+This plugin produces the `buildroot.lock` file in the result directory.  The
+lockfile describes the buildroot so it can be later reproduced by Mock (see the
+feature page, TODO).
+
+The BuildrootLock plugin is disabled by default, but automatically enabled with
+the `--calculate-build-dependencies` option.
+
+The file `buildroot.lock` is a JSON file.
+
+This plugin will not work correctly with `--offline` mode.
+
+## Format of the lockfile
+
+We currently do not provide a compatibility promise.  Only the very same version
+of Mock that produced the file is guaranteed to read and process it later (see
+the feature page, TODO).
+
+Example of such lockfile:
```

Comment:
It would be nice to provide JSON schema too. See my similar work in https://gitlab.com/fedora/legal/fedora-license-data/-/blob/main/tools/fedora-license-schema.json?ref_type=heads and validation https://gitlab.com/fedora/legal/fedora-license-data/-/blob/main/Makefile?ref_type=heads#L64

Reply:
Going to provide the schema-as-doc, and add a unit-test "testing" the schema.

Reply:
Still TBD